static/app/components/replays/canvasReplayerPlugin.tsx

```diff
@@ -212,69 +262,57 @@ export function CanvasReplayerPlugin(events: eventWithTime[]): ReplayPlugin {
         (node as HTMLCanvasElement).appendChild(el);
         containers.set(id, el);
       }
+
+      // See comments at definition of `handleQueue`
+      const queue = handleQueue.get(id);
+      handleQueue.set(id, []);
+      while (queue?.length) {
+        const queueItem = queue.shift();
+        if (!queueItem) {
+          return;
+        }
+        const [event, replayer] = queueItem;
+        processEvent(event, {shouldPreload: false, replayer});
+      }
```

Comment:
Logic is as follows:

* User seeks somewhere in the timeline
* below handler is called with the events before that spot in timeline with `isSync = true` for all the events
* create a list of canvas mutation events, indexed by its node id
* wait until `onBuild` is called with an indexed node id (this happens as rrweb is processing events normally)
* process events for the node id that was just built

Reply:
New Logic:

- User seeks somewhere in the timeline
- below handler is called with the events before that spot in timeline with isSync = true for all the events
- keep latest event, indexed by its node id
- we want to process the indexed events after all of the isSync events are handled. We do this by calling a debounced function that will attempt to process the latest event for each canvas id. we only remove from queue if event was successfully processed, otherwise ...
- wait until onBuild is called with an indexed node id (this happens as rrweb is processing events normally) a nd process events for the node id that was just built